src/main/java/org/jenkinsci/plugins/workflow/cps/CpsBodyExecution.java

```diff
@@ -172,29 +179,63 @@ private Continuable createContinuable(CpsThread currentThread, CpsCallableInvoca
     }
 
     @Override
-    public synchronized Collection<StepExecution> getCurrentExecutions() {
+    public Collection<StepExecution> getCurrentExecutions() {
```

Comment:
This method was not involved in the original bug, but it was wrong too, in the same way.